interfaces/Portalicious/src/app/components/registration-page-layout/registration-page-layout.component.ts

```diff
@@ -129,14 +147,22 @@ export class RegistrationPageLayoutComponent {
     return `${localized}${this.registration.data()?.registrationProgramId.toString() ?? ''} - ${this.registration.data()?.name ?? ''}`;
   });
 
-  readonly addNoteFormVisible = signal(false);
-
   readonly canUpdatePersonalData = computed(() =>
     this.authService.hasPermission({
       projectId: this.projectId(),
       requiredPermission: PermissionEnum.RegistrationPersonalUPDATE,
     }),
   );
+
+  readonly duplicateArray = computed(() => this.duplicates.data() ?? []);
+  readonly duplicateChipData = computed(() =>
+    getChipDataByDuplicateStatus(
+      this.duplicateArray().length > 0
```

Comment:
You could also get this information from: 
`this.registration.data().duplicateStatus`

as it also returns from: 
`/api/programs/{programId}/registrations/{id}`

Your solution also works.

But I think I would have:
- Used `this.registration.data().duplicateStatus` here
- And than just passed the referenceId to component app-registration-duplicates-banner (and than you may also not need to pass in loading anymore). So that one is more of a stand alone component with a single concern
- I am not completely confident about what the balance here is between seperation of concerns and keeping components dumb

So up to you :)

Reply:
Ha, I wasn't aware of `this.registration.data().duplicateStatus`. In that case I agree with you ✌️